### ISSUE TITLE
Manually install downloaded app update on next run

### ISSUE THREAD

**Issue by mvarakin** [electron-userland/electron-builder#2213]
Dear, 

I use electron-updater to update my app.
I can check for updates, force to download the update manually (thanks to autoDownload property) and install it automatically once it's been downloaded. 

Also, I can let a user desides if he wants to install the update right now or not.

In this case I can set a flag (ex. readyForInstall = true) once I got 'update-downloaded' event.
And as soon as a user wishes to install the update, I can check the flag and call quitAndInstall manually.
But it works in a frame of current app session only.

I can't understand how I can install the update on the next app run without downloading the same dist again? Docs say that quitAndInstall should be called only after 'update-downloaded' but this event never happens on the next run (as soon as I don't download the same update again).

**Comment by develar** [comment]
You cannot install downloaded file on next launch after app quit because file downloaded to temporary directory. What you can do — set autoDownload to false. If user agree — call download (other events will be called after it — e.g. update-downloaded). If not — do nothing.

**Comment by develar** [comment]
I doubt that you need to save download time. Deferred update install is strange — new version can be released and so on.

BTW, delta updates supported for nsis-web and will be supported for nsis next month.

**Comment by mvarakin** [comment]
Looking forward for delta updates then.
Thanks

**Comment by weiluenju** [comment]
> You cannot install downloaded file on next launch after app quit because file downloaded to temporary directory. What you can do — set autoDownload to false. If user agree — call download (other events will be called after it — e.g. update-downloaded). If not — do nothing.

Hi, @develar 

Is it possible to change the temporary directory?

I understand on macOS, the temporary directory will be cleared when macOS restart, so auto install update before next run is not possible.

But the application’s data directory in user’s directory will not be cleared when OS restart, for example:
```
Windows: %LOCALAPPDATA%/xxx
Linux: ~/.config/xxx
OS X: ~/Library/Application Support/xxx/Default
```
Maybe the updater can consider download the file into these directories.
